osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/AbstractGetPowerQualityProfileHandler.java

```diff
@@ -463,39 +439,105 @@ private ProfileEntryValueDto makeNumericProfileEntryValueDto(final DataObject da
             }
         } catch (final ProtocolAdapterException e) {
             LOGGER.error("Error creating ProfileEntryDto from {}", dataObject, e);
-            final String dbgInfo = this.dlmsHelper.getDebugInfo(dataObject);
-            return new ProfileEntryValueDto(dbgInfo);
+            final String debugInfo = this.dlmsHelper.getDebugInfo(dataObject);
+            return new ProfileEntryValueDto(debugInfo);
         }
     }
 
-    private List<ScalerUnitInfo> retrieveScalerUnits(final DlmsConnectionManager conn, final DlmsDevice device,
-            final List<GetResult> captureObjects) throws ProtocolAdapterException {
+    private Map<Integer, CaptureObjectDefinitionDto> createSelectableCaptureObjects(
+            final List<GetResult> captureObjects, List<String> logicalNames) throws ProtocolAdapterException {
 
-        final List<ScalerUnitInfo> result = new ArrayList<>();
+        Map<Integer, CaptureObjectDefinitionDto> selectableCaptureObjects = new HashMap<>();
 
+        // there is always only one GetResult
         for (final GetResult captureObjectResult : captureObjects) {
-            final DataObject dataObject = captureObjectResult.getResultData();
-            final List<DataObject> dataObjectList1 = dataObject.getValue();
-            for (final DataObject captureObjectDataObject : dataObjectList1) {
+
+            final List<DataObject> dataObjects = captureObjectResult.getResultData().getValue();
+            int positionInDataObjectsList = 0;
+
+            for (DataObject dataObject : dataObjects) {
```

Comment:
Consider using classic for-loop with variable positionInDataObjectsList inlined; this ensures increment of the variable and improves readability

Reply:
you are right, that would be an improvement.